src/SprykerEco/Zed/Unzer/Persistence/UnzerRepository.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Persistence;
+
+use Generated\Shared\Transfer\MerchantUnzerParticipantTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransactionTransfer;
+use Generated\Shared\Transfer\PaymentUnzerTransfer;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+use SprykerEco\Zed\Unzer\Persistence\Mapper\UnzerPersistenceMapper;
+
+/**
+ * @method \SprykerEco\Zed\Unzer\Persistence\UnzerPersistenceFactory getFactory()
+ */
+class UnzerRepository extends AbstractRepository implements UnzerRepositoryInterface
+{
+    /**
+     * @param string $merchantReference
+     *
+     * @return \Generated\Shared\Transfer\MerchantUnzerParticipantTransfer|null
+     */
+    public function findMerchantUnzerParticipantByMerchantReference(string $merchantReference): ?MerchantUnzerParticipantTransfer
+    {
+        /** @var \Orm\Zed\Unzer\Persistence\SpyMerchantUnzerParticipant|null $merchantUnzerParticipantEntity */
+        $merchantUnzerParticipantEntity = $this->getFactory()->createMerchantUnzerParticipantQuery()
+            ->useMerchantQuery()
+                ->filterByMerchantReference($merchantReference)
+            ->endUse()
+            ->findOne();
+
+        if ($merchantUnzerParticipantEntity === null) {
+            return null;
+        }
+
+        return $this->getFactory()->createUnzerPersistenceMapper()
+            ->mapMerchantUnzerParticipantEntityToMerchantUnzerParticipantTransfer(
+                $merchantUnzerParticipantEntity,
+                new MerchantUnzerParticipantTransfer()
+            );
+    }
+
+    /**
+     * @param string $orderReference
+     *
+     * @return \Generated\Shared\Transfer\PaymentUnzerTransfer|null
+     */
+    public function findPaymentUnzerByOrderReference(string $orderReference): ?PaymentUnzerTransfer
+    {
+        $paymentUnzerEntity = $this->getFactory()->createPaymentUnzerQuery()
+            ->filterByOrderId($orderReference)
+            ->findOne();
+
+        if ($paymentUnzerEntity === null) {
+            return null;
+        }
+
+        return $this->getFactory()->createUnzerPersistenceMapper()->mapPaymentUnzerEntityToPaymentUnzerTransfer(
+            $paymentUnzerEntity,
+            new PaymentUnzerTransfer()
+        );
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return \Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer
+     */
+    public function findPaymentUnzerOrderItemCollectionByOrderId(string $orderId): PaymentUnzerOrderItemCollectionTransfer
```

Comment:
Should be `get` because cannot return `null`?
`findPaymentUnzerOrderItemCollectionByOrderId()` -> `getPaymentUnzerOrderItemCollectionByOrderId()`